src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutes.scala

```diff
@@ -35,44 +42,44 @@ abstract class LeoRoutes(val leonardoService: LeonardoService, val proxyService:
     }
 
   def leoRoutes: Route =
-    path("cluster" / Segment / Segment) { (googleProject, clusterName) =>
-      requireUserInfo { userInfo =>
-        put {
-          entity(as[ClusterRequest]) { cluster =>
-            complete {
-              leonardoService.createCluster(GoogleProject(googleProject), ClusterName(clusterName), cluster).map { cluster =>
-                StatusCodes.OK -> cluster
+    requireUserInfo { userInfo =>
+      if (whitelist.contains(userInfo.userEmail.value)) {
```

Comment:
I don't think putting an if statement inside the route is the correct approach. It might be better to add a new directive which checks if the user is in the whitelist. So the route would look like:
```
path("cluster" / Segment / Segment) { (googleProject, clusterName) =>
  requireUserInfo { userInfo =>
    checkWhitelist(userInfo) {
      path(...) { /* etc */ }
    }
  }
}
```

`checkWhitelist` would be defined as a `Directive0` which checks the user against the whitelist and either fails the route or passes it through. 

An analog is the directive that we have for checking if a user is an admin. E.g.: 
https://github.com/broadinstitute/sam/blob/develop/src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserInfoDirectives.scala#L18

Reply:
On second thought, enh. This is essentially throwaway code right? If it works we can just leave it. We can add the directive when we do real auth later.